File: .agent-app/frontend/src/types/portfolio.ts
/**
 * Portfolio model types based on backend app/models/portfolio.py
 */

export interface PortfolioCreate {
  name: string;
  description?: string | null;
  status?: string;
  owner_id: number;
}

export interface PortfolioUpdate {
  name: string;
  description?: string | null;
  status?: string | null;
  owner_id: number;
}

export interface PortfolioResponse {
  id: number;
  name: string;
  description?: string | null;
  status: string;
  created_at: string;
  updated_at: string;
  is_active: boolean;
}

File: .agent-app/backend/app/models/portfolio.py
from pydantic import BaseModel
from typing import Optional

class PortfolioCreate(BaseModel):
    name: str
    description: Optional[str] = None
    status: Optional[str] = 'active'
    owner_id: int

class PortfolioUpdate(BaseModel):
    name: str
    description: Optional[str] = None
    status: Optional[str] = None
    owner_id: int

class PortfolioResponse(BaseModel):
    id: int
    name: str
    description: Optional[str] = None
    status: str
    created_at: str
    updated_at: str
    is_active: bool
